### PR TITLE
Fix searching by genre/language not working in beatmap listing overlay

### PR DESCRIPTION
**Pull request by EVAST9919**

### CODE REVIEW

**Review by bdach — APPROVED** [review]
seems reasonable 👍 

Still not really a fan of the number of ctor parameters on `SearchBeatmapSetsRequest` but it hasn't changed here so...

File: osu.Game/Overlays/BeatmapListing/BeatmapListingFilterControl.cs [hunk 1]
```diff
@@ -186,6 +188,9 @@ private void performRequest()
                 if (sets.Count == 0)
                     noMoreResults = true;
 
+                if (CurrentPage == 0)
+                    searchControl.BeatmapSet = sets.FirstOrDefault();
```
Comment (peppy) [inline osu.Game/Overlays/BeatmapListing/BeatmapListingFilterControl.cs:192]: This **really** should have been in its own PR. Please don't mix fixes together again.

**Comment by peppy** [comment]
Parameter count kinda has changed here. Really should be doing something like [FilterCriteria](https://github.com/ppy/osu/blob/master/osu.Game/Screens/Select/FilterCriteria.cs).

**Review by peppy — APPROVED** [review]
Required changes can be made in a follow-up PR.

**Pull request merged**